algorithms/src/snark/marlin/ahp/matrices.rs

```diff
@@ -213,11 +220,14 @@ pub(crate) fn arithmetize_matrix<F: PrimeField>(label: &str, matrix_evals: Matri
 
     end_timer!(matrix_time);
 
+    let mut labels: VecDeque<String> =
```

Comment:
do we need a `VecDeque` here? seems to me like we could leave it w/o the `collect` and then call `next()` instead of `pop_front()` below

Reply:
Brilliant!!! f2ede7000